### PR TITLE
Add $allOf object inheritance

### PR DESCRIPTION
**Pull request by homestar9**
This is a custom behavior for allowing schema inheritance.

### CODE REVIEW

**Comment by homestar9** [comment]
There's a problem executing the tests as they are written with this PR because of this line on 310 
```
return ReferenceDocument.getNormalizedDocument();
```

If I don't normalize the document, all of the CFC metadata gets included instead of the document struct.  I started trying to rewrite the tests, but started going down a rabbit hole and didn't want to rewrite all the tests from scratch. Maybe you have some better ideas on how to handle the `ReferenceDocument` and when to call `getNormalizedDocument()` to ensure the existing tests continue to pass.

I wanted to add some tests for the actual inheritance processing.  My idea was to change /tests/resources/petstore/petstore.json to this:
```
{
	"openapi": "3.0.0",
	"info": {
		"version": "1.0.0",
		"title": "Swagger Petstore",
		"license": {
			"name": "MIT"
		}
	},
	"servers": [
		{
			"url": "http://petstore.swagger.io/v1"
		}
	],
	"paths": {
		"/pets": {
      		"$ref" : "paths/pets.json#/pets"
		},
		"/pets/{petId}": {
			"$ref" : "paths/pets.json#/pets/{petId}"
		}
	},
    "components" : {
        "schemas": {
            "$ref": "paths/definitions/Schemas.json"
        }
    }
}
```

The paths/definitions/Schemas.json file would look like this:
```
{
    "Response": { "$ref":"Response.json" },
    "Pagination": { "$ref":"Pagination.json" },
    "Media": { "$ref": "Media.json" },
    "MediaBook": { "$ref": "MediaBook.json" }
    "MediaMusic": { "$ref": "MediaMusic.json" }
}
```

The /paths/definitions/Response.json would look like this:
```
{
    "type": "object",
    "properties": {
        "error": {
            "description": "Flag to indicate an error.",
            "type": "boolean"
        },
        "messages": {
            "description": "An array of messages related to the request.",
            "type": "array",
            "items": {
                "type": "string"
            }
        },
        "data": {
            "description": "The data packet",
            "type": "object",
            "properties": {}
        }
    }
}
```

The /paths/definitions/Media.json would look like this:
```
{
    "type": "object",
    "properties": {
        "id": {
            "type": "string"
        },
        "title": {
            "type": "string"
        }
    }
}
```

And finally, the /paths/definitions/MediaBook.json (subclass schema) would look like this:
```
{
    "allOf": [
        { "$ref": "#/components/schemas/Media" },
        {
            "type": "object",
            "properties": {
                "Author": {
                    "type": "string"
                },
                "ISBM": {
                    "type": "string"
                }
            }
        }
    ]
}
```
... I would also create similar schema JSON files for `MediaMusic` and `Pagination`

Once that was all set up, I wanted to create a new path that would return the response object extended by the MediaBook object.  It might look something like this:

```
{
  "/media/": {
    "get": {
      "summary": "Returns a list of media objects",
      "operationId": "index",
      "tags": [
        "media"
      ],
      "parameters": [
        {
          "name": "title",
          "in": "query",
          "required": false,
          "schema": {
            "type": "string"
          }
        }
      ],
      "responses": {
        "200": {
          "description": "A page of media objects",
          "content": {
            "application/json": {
              "schema": {
                // This is where we get to test our special swagger-sdk inheritance!
                 "$allOf": [
                  {
                    "$ref": "../../_schemas/response.json"
                  },
                  {
                    "type": "object",
                    "properties": {
                      "data": {
                        "type": "object",
                        "properties": {
                          "results": {
                            "type": "array",
                            "items": {
                              "anyOf": [
                                {
                                  "$ref": "#/components/schemas/MediaBook"
                                },
                                {
                                  "$ref": "#/components/schemas/MediaMusic"
                                }
                              ]
                            }
                          },
                          "pagination": {
                            "$ref": "#/components/schemas/Pagination"
                          }
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      }
    }
  }
}
```

I know the above is a lot, so let me know if my intent doesn't make any sense and I will try to clarify.

**Comment by homestar9** [comment]
Note: I forgot to pull `development` first on my fork.  I will update the PR shortly.

**Comment by jclausen** [comment]
Thanks @homestar9 !  I will merge this and then update the tests for this functionality.

File: models/OpenAPI/Parser.cfc [hunk 1]
```diff
@@ -202,23 +276,23 @@ component name="OpenAPIParser" accessors="true" {
 			//Files receive a parser reference
 			if( left( FilePath, 4 ) == 'http'  ){
 
-				var ReferenceDocument = Wirebox.getInstance( "OpenAPIParser@SwaggerSDK" ).init(  $ref );
+				ReferenceDocument = Wirebox.getInstance( "OpenAPIParser@SwaggerSDK" ).init(  $ref );
 
 			} else if( len( FilePath ) && fileExists( getDirectoryFromPath( getBaseDocumentPath() ) &  FilePath )){
 
-				var ReferenceDocument = Wirebox.getInstance( "OpenAPIParser@SwaggerSDK" ).init(  getDirectoryFromPath( getBaseDocumentPath() ) & $ref );
+                ReferenceDocument = Wirebox.getInstance( "OpenAPIParser@SwaggerSDK" ).init(  getDirectoryFromPath( getBaseDocumentPath() ) & FilePath );
```
Comment (jclausen) [inline models/OpenAPI/Parser.cfc:283]: This line change actually, broke the `$ref` inheritance functionality.  I've reverted it back to use `$ref`

**Pull request merged**